code/modules/reagents/reagent_containers/syringes.dm

```diff
@@ -20,13 +21,69 @@
 	sharp = 1
 	noslice = 1
 	unacidable = 1 //glass
-	var/mode = SYRINGE_DRAW
+	var/mode = SYRINGE_CAPPED
+	var/used = FALSE
+	var/dirtiness = 0
+	var/list/targets
+	var/list/datum/disease2/disease/viruses
 	var/image/filling //holds a reference to the current filling overlay
 	var/visible_name = "a syringe"
 	var/time = 30
 	center_of_mass = null
 	drop_sound = 'sound/items/drop/glass.ogg'
 
+/obj/item/weapon/reagent_containers/syringe/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/item/weapon/reagent_containers/syringe/Destroy()
+	LAZYCLEARLIST(viruses)
+	LAZYCLEARLIST(targets)
+	return ..()
+
+/obj/item/weapon/reagent_containers/syringe/process() // this only happens once it's used
+	if(prob(75)) // sorry, had to nerf this.
+		return
+	dirtiness = min(dirtiness + LAZYLEN(targets), 75)
+	if(dirtiness >= 75)
+		STOP_PROCESSING(SSprocessing, src)
+	return 1
+
+/obj/item/weapon/reagent_containers/syringe/proc/infect_limb(var/obj/item/organ/external/eo)
+	eo.germ_level += INFECTION_LEVEL_ONE+30
+
+/obj/item/weapon/reagent_containers/syringe/proc/dirty(var/mob/living/carbon/human/target, var/obj/item/organ/external/eo)
+	LAZYINITLIST(targets)
+
+	//We can't keep a mob reference, that's a bad idea, so instead name+ref should suffice.
+	var/hash = md5(target.real_name + "\ref[target]")
+
+	//Just once!
+	targets |= hash
+
+	//Grab any viruses they have
+	if(LAZYLEN(target.virus2.len))
+		LAZYINITLIST(viruses)
+		var/datum/disease2/disease/virus = pick(target.virus2.len)
+		viruses[hash] = virus.getcopy()

```

Comment:
Why are we using two lists to hold hashes...? Specially since both hold each hash only once.

Reply:
What? Are you talking about target.virus2? That's because that's on an individual target, of which there can be multiple.